commands/download.go

```diff
@@ -1,14 +1,97 @@
 package commands
 
 import (
+	"errors"
 	"fmt"
-	"github.com/jfrog/jfrog-cli-core/plugins/components"
+	"github.com/jfrog/jfrog-cli-core/artifactory/utils"
 	"github.com/jfrog/jfrog-cli-core/utils/config"
+	"github.com/jfrog/jfrog-client-go/utils/io/fileutils"
+	"github.com/jfrog/jfrog-client-go/utils/io/httputils"
 	"github.com/jfrog/jfrog-client-go/utils/log"
+	"io/ioutil"
+	"net/http"
+	"time"
 )
 
-func downloadSupportBundle(ctx *components.Context, rtDetails *config.ArtifactoryDetails, conf *supportBundleCommandConfiguration, response creationResponse) (string, error) {
-	tmpFile := "/tmp/foo.tmp"
-	log.Debug(fmt.Sprintf("Download Support Bundle %s from %s to %s", response.Id, rtDetails.Url, tmpFile))
+type httpGetExecutor interface {
+	SendGet(url string, followRedirect bool, httpClientsDetails *httputils.HttpClientDetails) (resp *http.Response, respBody []byte, redirectUrl string, err error)
+}
+
+func downloadSupportBundle(timeout time.Duration, retryInterval time.Duration, rtDetails *config.ArtifactoryDetails, bundleId string) (string, error) {
+	dirPath, err := fileutils.CreateTempDir()
+	if err != nil {
+		return "", err
+	}
+	tmpFile := fmt.Sprintf("%s/%s", dirPath, bundleId)
+	log.Debug(fmt.Sprintf("Download Support Bundle %s from %s to %s", bundleId, rtDetails.Url, tmpFile))
+
+	manager, err := utils.CreateServiceManagerWithProgressBar(rtDetails, 1, false, nil)
+	if err != nil {
+		return "", err
+	}
+
+	httpClientDetails := manager.GetConfig().GetServiceDetails().CreateHttpClientDetails()
+
+	url := fmt.Sprintf("%sapi/system/support/bundle/%s/archive", rtDetails.GetUrl(), bundleId)
+	resp, body, _, err := manager.Client().SendGet(url, true, &httpClientDetails)
+	if err != nil {
+		return "", err
+	}
+
+	log.Debug(fmt.Sprintf("Got %s", resp.Status))
+
+	if resp.StatusCode == http.StatusNotFound {
+		log.Debug("Support Bundle not found, will retry")
+		body, err = retry(manager.Client(), url, httpClientDetails, timeout, retryInterval)
+		if err != nil {
+			return "", err
+		}
+	}
+
+	if resp.StatusCode != http.StatusOK {
+		return "", fmt.Errorf("http request failed with: %s", resp.Status)
+	}
+
+	err = ioutil.WriteFile(tmpFile, body, 0644)
+	if err != nil {
+		return "", err
+	}
+
 	return tmpFile, nil
 }
+
+func retry(executor httpGetExecutor, url string, httpClientDetails httputils.HttpClientDetails, timeout time.Duration, retryInterval time.Duration) ([]byte, error) {
+	ticker := time.NewTicker(retryInterval)
+	defer ticker.Stop()
+	done := make(chan bool)
+	go func() {
+		time.Sleep(timeout)
+		done <- true
+	}()
+
+	for {
+		select {
+		case <-done:
+			return nil, errors.New("timeout trying to download the support bundle")
+		case <-ticker.C:
+			log.Debug("Attempting to download Support Bundle")
+			resp, body, _, err := executor.SendGet(url, true, &httpClientDetails)
```

Comment:
We should make sure that the download is not possible until the SB is ready. Otherwise, we should probably call `GET system/support/bundle/{bundle_id}` and check the bundle status.

Reply:
I don't think it is possible to get a partial SB, but I'll go with checking the status first just in case. In this case I won't retry the download on 404.